examples/nest/hpc_benchmark_invariant.sli

```diff
@@ -0,0 +1,512 @@
+    /E_detector /spike_detector Create def
+    E_detector
+    <<
+       /withtime true
+       /to_file true
+       /label detector_label
+    >> SetStatus
+  } if
+
+  memory_thisjob cvs ( # virt_mem_after_nodes) join logger /log call
+
+  toc /BuildNodeTime Set
+
+  BuildNodeTime cvs ( # build_time_nodes) join logger /log call
+
+  tic
+
+  % Create custom synapse types with appropriate values for
+  % our excitatory and inhibitory connections
+  /static_synapse_hpc << /delay delay >> SetDefaults
+  /static_synapse_hpc /syn_std  CopyModel
+  /static_synapse_hpc /syn_ex << /weight JE_pA >> CopyModel
+  /static_synapse_hpc /syn_in << /weight JE_pA g mul >> CopyModel
+
+  stdp_params /weight JE_pA put 
+  /stdp_pl_synapse_hom_hpc stdp_params SetDefaults
+
+
+  M_INFO (BuildNetwork)
+  (Connecting stimulus generators.) message
+
+  % Connect Poisson generator to neuron
+
+  E_stimulus E_stimulus cvgidcollection E_neurons << /rule (all_to_all) >> << /model /syn_ex >> Connect
+  E_stimulus E_stimulus cvgidcollection I_neurons << /rule (all_to_all) >> << /model /syn_ex >> Connect
+
+  M_INFO (BuildNetwork)
+  (Connecting excitatory -> excitatory population.) message
+
+  E_neurons E_neurons << /rule (fixed_indegree) /indegree CE /autapses false /multapses true >>
+                      << /model /stdp_pl_synapse_hom_hpc >> Connect
+
+  M_INFO (BuildNetwork)
+  (Connecting inhibitory -> excitatory population.) message
+
+  I_neurons E_neurons << /rule (fixed_indegree) /indegree CI /autapses false /multapses true >>
+                      << /model /syn_in >> Connect
+
+  M_INFO (BuildNetwork)
+  (Connecting excitatory -> inhibitory population.) message
+
+  E_neurons I_neurons << /rule (fixed_indegree) /indegree CE /autapses false /multapses true >>
+                      << /model /syn_ex >> Connect
+
+  M_INFO (BuildNetwork)
+  (Connecting inhibitory -> inhibitory population.) message
+
+  I_neurons I_neurons << /rule (fixed_indegree) /indegree CI /autapses false /multapses true >>
+                      << /model /syn_in >> Connect
+
+
+  record_spikes true eq
+  {
+    E_neurons size Nrec lt
+    {
+      M_ERROR (BuildNetwork)
+      (Networks is smaller than number of neurons to record from!) message
+      1 quit_i
+    } if
+
+    M_INFO (BuildNetwork)
+    (Connecting spike detectors.) message
+
+    /recordees E_from E_from Nrec add 1 sub cvgidcollection def
+    /detector E_detector E_detector cvgidcollection def
+    recordees detector << /rule (all_to_all) >> << /model /syn_std >> Connect
+  } if  
+
+  % read out time used for building
+  toc /BuildEdgeTime Set
+
+  BuildEdgeTime cvs ( # build_edge_time ) join logger /log call
+  memory_thisjob cvs ( # virt_mem_after_edges) join logger /log call
+
+ } def % end of buildnetwork
+
+/RunSimulation
+{
+
+  % open log file
+  log_file logger /init call
+
+  ResetKernel
+
+  memory_thisjob cvs ( # virt_mem_0) join logger /log call
+
+  BuildNetwork
+
+  tic
+
+  Prepare
```

Comment:
why use prepare, run, cleanup instead of simulate? do the advantages outweigh the change wrt the hpc_benchmark?

Reply:
I think I must have been inspired by another benchmark. Tested quickly now, and it does not seem to have an effect, will change to Simulate.